### PR TITLE
[Api-draft] Add seatunnel kafka source connectors 

### PR DESCRIPTION
**Pull request by Hisoka-X**
## Purpose of this pull request
Add seatunnel kafka source connectors

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by ruanwenjun** [comment]
@BenJFan Please resolve the conflicts.

**Comment by Hisoka-X** [comment]
> @BenJFan Please resolve the conflicts.

Done

File: seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/execution/FlinkTaskExecution.java [hunk 1]
```diff
@@ -46,7 +47,8 @@ public class FlinkTaskExecution {
 
     public FlinkTaskExecution(Config config) {
         this.config = config;
-        this.flinkEnvironment = createFlinkEnvironment();
+        this.flinkEnvironment = (FlinkEnvironment) new EnvironmentFactory<>(config, EngineType.FLINK).getEnvironment();
```
Comment (ruanwenjun) [inline seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/execution/FlinkTaskExecution.java:50]: Please rollback this file.

File: seatunnel-core/seatunnel-spark-starter/src/main/java/org/apache/seatunnel/core/starter/spark/execution/SparkTaskExecution.java [hunk 2]
```diff
@@ -45,7 +44,8 @@ public class SparkTaskExecution {
 
     public SparkTaskExecution(Config config) {
         this.config = config;
-        this.sparkEnvironment = getSparkEnvironment(config);
+        this.sparkEnvironment = (SparkEnvironment) new EnvironmentFactory<>(config, EngineType.SPARK).getEnvironment();
```
Comment (ruanwenjun) [inline seatunnel-core/seatunnel-spark-starter/src/main/java/org/apache/seatunnel/core/starter/spark/execution/SparkTaskExecution.java:47]: Please rollback this file.

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/execution/FlinkTaskExecution.java (hunk 1 above)
Comment (Hisoka-X) [reply seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/execution/FlinkTaskExecution.java:50]: I change the EnvironmentFactory and FlinkEnvironment to support change `job.mode` and create `StreamExecutionEnvironment`. So this method will work, this way have any problem?

Comment (ruanwenjun) [reply seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/execution/FlinkTaskExecution.java:50]: OK, I see your change in `EnvironmentFactory`, the old `EnvironmentFactory` has some problem in new API.

**Pull request merged**